src/v8/isolate.rs

```diff
@@ -237,6 +242,11 @@ impl V8Isolate {
     pub fn cancel_terminate_execution(&self) {
         unsafe { v8_CancelTerminateExecution(self.inner_isolate) }
     }
+
+    /// Returns a raw pointer to a [v8_isolate].
+    pub fn get_raw(&self) -> *mut v8_isolate {
```

Comment:
Why do we need this API? It expose an internal pointer, I would prefer that any api that require internal pointer will simply get `&V8Isolate` if its possible.

Reply:
I guess you need it for the inspector? Why not split the inspect API to local and persisted like the V8Local and Persisted value? In order to re-enter the inspector you need to provide an isolate scope, we can also check then for the isolate ID. I guess this way we will have no need for the `get_raw` function.

Reply:
Maybe the local and persisted idea is confusing, but my point is that we should only use the inspector when the isolate is locked. Locking an isolate is indicated by isolate_scope. So why not a lock like API for the inspector such that in order to interact with the inspector (send messages for example), the user need to provide an isolate scope (a proof that the isolate is locked). Then the user get back and inspector guard which can be used to access the inspector and can not outlive the isolate scope.
Also, when getting the inspector guard, we can check the inspector ID to make sure we are using it with the right isolate.
WDYT?